### PR TITLE
Changes to VAO to reuse VAO iD and VBO Ids and allow interleaving of …

### PR DESCRIPTION
**Pull request by Vittalbabu**
Changes to VAO to reuse VAO iD and VBO Ids and allow interleaving of the 3 builtin attributes.

### CODE REVIEW

**Comment by NolaDonato** [comment]
The change looks good. I tested with multilight and jassimp samples so it works with skinning and lighting together.

My only comment is that we should go further and make all the vertex attributes interleaved. We need to separate vertices and indices for X3D. Perhaps we can interleave all the vertex attributes at that time.

**Comment by Vittalbabu** [comment]
Making all vertex attributes interleaved is in progress.

File: GVRf/Framework/framework/src/main/jni/objects/mesh.cpp [hunk 1]
```diff
@@ -208,61 +208,120 @@ void Mesh::bindVertexAttributes(GLuint programID)
...
+    if (attribSize == 0) {
         std::string error = "no vertex data yet, shouldn't call here. ";
         throw error;
         return;
     }
 
-    glGenVertexArrays(1, &vaoID_);
+    if (vao_dirty_ || vaoID_ == GVR_INVALID)
+    {
+        glGenVertexArrays(1, &vaoID_);
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/jni/objects/mesh.cpp]: In case of dirty, should the vaoId be glDeleteVertexArrays-ed?

**Comment by liaxim** [comment]
@Vittalbabu Should this be closed?

**Comment by Vittalbabu** [comment]
This can be closed.

**Pull request closed without merging**